src/server/faculty/faculty.service.ts

```diff
@@ -36,4 +47,69 @@ export class FacultyService {
       .orderBy('"displayName"')
       .getRawMany();
   }
+
+  public async updateFacultyAbsences(
+    absenceReqInfo: Pick<Absence, 'id' | 'type'>
+  ): Promise<Absence> {
+    const existingAbsence = await this.absenceRepository
+      .findOne({
+        relations: ['faculty', 'semester'],
+        where: { id: absenceReqInfo.id },
+      });
+
+    const updateQuery = this.absenceRepository.createQueryBuilder()
+      .update(Absence);
+
+    // Are we going to have to do NO_LONGER_ACTIVE gymnastics?
+    if (
+      [existingAbsence.type, absenceReqInfo.type]
+        .some((value) => value === ABSENCE_TYPE.NO_LONGER_ACTIVE)
```

Comment:
Minor note: This could be `.includes(ABSENCE_TYPE.NO_LONGER_ACTIVE)`. As far as I can tell it's effectively the same, but _slightly_ easier to read while going over the code.

Reply:
I think I agree, fixed in cb002aaa5124533a228f6be2d65326816f4d37cd